### PR TITLE
ansible: make use of role dependencies

### PR DESCRIPTION
**Pull request by AshCripps**
noticed the other day that the `java-base` role calls package upgrade and makes it run twice so remove that and also make use of proper dependencies within ansible to clean it up a bit.

Also split out user creation to its own role to ensure the build-v8 role still works.

needs some more testing before merging.

Is there any other parts of the playbooks I should "clean up" whilst im at it @nodejs/build ?

### CODE REVIEW

File: ansible/roles/user-create/vars/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  ibmi: 'ibmi73',
+  macos: 'macos',
+  rhel7: 'rhel7',
+  systemd: ['centos7', 'debian8', 'debian9', 'debian10', 'fedora', 'ubuntu1604', 'ubuntu1804'],
+  svc: 'smartos',
+  upstart: ['ubuntu12', 'ubuntu1404'],
+  zos_start: 'zos'
+  }
+
+init_type: "{{ os|match_key(init, raise_error=False)|default(os|stripversion|match_key(init, raise_error=False)) }}"
```
[the remark this answers is not in the document]
Comment (AshCripps) [reply ansible/roles/user-create/vars/main.yml:21]: I wasnt 100% sure, are they used to map the os's in the conditional blocks?

[the remark this answers is not in the document]
Comment (AshCripps) [reply ansible/roles/user-create/vars/main.yml:21]: dont think its needed removing

[the remark this answers is not in the document]
Comment (richardlau) [reply ansible/roles/user-create/vars/main.yml]: It's used to lookup/determine how to start the Jenkins agent
https://github.com/nodejs/build/blob/38132df5a722e8c408129a41065c6ac140434106/ansible/roles/jenkins-worker/vars/main.yml#L78

**Comment by richardlau** [comment]
FWIW "dependencies" is misspelled in the commit/PR title.

**Comment by AshCripps** [comment]
@richardlau corrected typo

**Review by richardlau — APPROVED** [review]

**Pull request merged**